mvebu: convert to DSA

dengqf6 opened the pull request:
Convert Linksys WRT series and SolidRun Clearfog to DSA

Cc: @hauke

Commit:
mvebu: drop swconfig patches

Drop swconfig patches for Clearfog and Linksys WRT

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
mvebu: remove swconfig symbols from kernel config

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
mvebu: update default config for DSA

Update network/LED configuration for DSA driver.
sysupgrade from images prior to this commit with config preserved
will break the ethernet.

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
mvebu: drop WiFi migration script

Not used in kernel 5.4

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
mvebu: use ucidef to set up MAC address

Use ucidef to set up MAC address instead of preinit script

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
mvebu: drop 06_set_iface_mac preinit script

MAC address is set in board.d script
Interface swapping is not needed anymore as switching to DSA breaks
previous configuration anyway

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
generic: backport mv88e6xxx port mirroring support

Backport port mirroring support for mv88e6xxx

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

Commit:
mvebu: remove swconfig package

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

adrianschmutzler commented on target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network:
```diff
@@ -4,49 +4,82 @@
+		label_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
+		lan_mac=$label_mac
+		wan_mac=$label_mac
+		;;
```
Missing esac?

adrianschmutzler commented:
Hi, since it's more a transfer that two separate changes, I'd merge commits

1. mvebu: use ucidef to set up MAC address
2. mvebu: drop 06_set_iface_mac preinit script

into one. That should also make reviewing easier.

Note that the MAC address adjustment done by preinit and ucidef has different mechanisms, however, I haven't checked so far what happens in detail when DSA is used.

adrianschmutzler commented:
For mamba, wan port is labelled "internet" in kernel. Don't think this is patched here or did I overlook it (would favor a kernel patch instead of having a separate entry in board.d).

adrianschmutzler commented:
Can you be a bit more specific on why the wifi script is removed?

dengqf6 commented:
> Can you be a bit more specific on why the wifi script is removed?

It's for LEDE 17.01. Do we still have to keep that?

adrianschmutzler commented:
> > Can you be a bit more specific on why the wifi script is removed?
> 
> It's for LEDE 17.01. Do we still have to keep that?

Ah, I'm too ath79-focussed, wasn't aware that mvebu is on 4.14 already in 18.06.

Personally, I'd be fine with removing it then. However, other members might see this differently.

Since this does not seem related to DSA switch to me, maybe send this patch to the list with a more specific commit message and my Acked-by. If nobody cries during next week, I'll apply it.

adrianschmutzler commented on target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network:
```diff
@@ -4,49 +4,83 @@
+	buffalo,ls421de)
+		lan_mac=$(mtd_get_mac_ascii u-boot-env eth1addr)
+		;;
+	linksys,caiman|linksys,cobra|linksys,rango|linksys,shelby|linksys,venom)
```
Nitpick: I'd prefer to have this linewise like above:
```
	linksys,caiman|\
	linksys,cobra|\
	linksys,rango|\
	linksys,shelby|\
	linksys,venom)
```

neheb commented:
![240d22bf3aeda750fca5c00d2dfbf72a](https://user-images.githubusercontent.com/2918873/79699101-0f0a3100-8242-11ea-98b1-6f26c06a5dba.jpg)

neheb commented on target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network:
```diff
@@ -36,11 +34,7 @@ solidrun,clearfog-pro-a1)
-	# switch port 5 is connected to eth1
-	ucidef_add_switch "switch0" \
-		"0:lan:5" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1" "5u@eth1" "6:lan:6"
+	ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4 lan5 lan6" "eth0 eth2"
```
I'm confused on this. Is this set up like the Turris Omnia? That is, 5 switched ports and one dedicated one. If so, I would set it up similar to that.

neheb replied:
Just read the comments. Never mind.

lucenera commented:
You are the deus ex machina!  You are turning my quarantine. After tinkering with the 5.4 kernel patch (now finally included in master), you arrive with these new delights. Thank you.

lucenera commented:
I built with DSA.  But how do you configure it now? I have no idea. Are there guides on this? My setup is simple, I just need a tagged vlan on the WAN to connect with the ISP.

dengqf6 commented:
> I just need a tagged vlan on the WAN to connect with the ISP.

Use `wan.vid` as wan's interface

lucenera commented:
> > I just need a tagged vlan on the WAN to connect with the ISP.
> 
> Use `wan.vid` as wan's interface

For this thing, however very simple, it worked a marvel! Thanks a lot! Even easier to set up.

anomeome commented:
Appears wan port is not found on _mamba_ [initialisation](https://termbin.com/x11n)

adrianschmutzler commented:
> For mamba, wan port is labelled "internet" in kernel.

This still needs to be fixed. Either by changing DTS in kernel, or by using a separate case in 02_network. I just started a general discussion about what should be the preferred way on the devel mailing list.

neheb commented:
I vote for the former.

kkudielka reviewed (commented):
EDIT: Sorry, please disregard this comment.

anomeome commented:
Yes, I saw your post above, but wanted to add the datapoint, and yes, changing the DTS for the _mamba_ from _internet_ to _wan_ appears to get the job done.

Was there any consenus reached in the 5.4 PR regarding
`300-mvneta-tx-queue-workaround.patch`
and the movement to DSA?

njhsi commented:
question: 

  in the ifconfig command output, all eth0/1.XX are gone, but still **eth1** left. is this right? not good for OCD people I guess..

----------------
**eth1**      Link encap:Ethernet  HWaddr XE:X3:81:X8:X8:XX
          UP BROADCAST RUNNING MULTICAST  MTU:1508  Metric:1
          RX packets:780384 errors:0 dropped:0 overruns:0 frame:0
          TX packets:184367 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1024
          RX bytes:289682460 (276.2 MiB)  TX bytes:4275045339 (3.9 GiB)
          Interrupt:38

schm0 commented:
On cobra, eth0 gets disabled at boot and eth1 is used for all interfaces lan1-4,wan.
The problem is eth0 has the correct mac address while eth1 (and all subinterfaces) get a random mac address assigned.

adrianschmutzler commented:
> On cobra, eth0 gets disabled at boot, and eth1 is used for all interfaces lan1-4, wan.
> The problem is eth0 has the correct mac address while eth1 (and all subinterfaces) get a random mac address assigned.

Though this smells like a workaround, maybe you can just set the eth1 MAC address from flash?

Commit:
mvebu: rename Linksys Mamba WAN port

Rename it to wan to match Linksys Armada 385 series

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

schm0 commented:
Atm I configured the mac addresses in /etc/config/network.

How does DSA selects which CPU Port gets used?

I think it's possible to swap the eth ports in the corresponding DTS file. But that only fixes the naming of the interfaces and not the random mac address issue.
I wonder why the DSA driver selects the "wrong" CPU Port..

dengqf6 commented:
@schm0 
eth0 connects to the switch port@6, to use it, you should also change `port@5` to `port@6`

schm0 commented:
Thank you very much that worked!

Commit:
mvebu: use eth0 as DSA CPU port for Linksys WRT

eth0 has HW MAC address while eth2 does not.
Use eth0 instead so we don't have to set LAN MAC manually.
Disable unused eth2, until multi CPU port is supported.

Signed-off-by: DENG Qingfang <dengqf6@mail2.sysu.edu.cn>

dengqf6 commented:
@anomeome Does Linksys Mamba set eth0/1 MAC address correctly?

schm0 commented:
According to the wiki all devices use CPU Port 6 for eth0 expect WRT1900AC v1.
https://oldwiki.archive.openwrt.org/toh/linksys/wrt_ac_series

But is/are there any disadvantage(s)?

anomeome commented:
Re: _mamba_. The _lanx_ and _wan_ devices have the correct, same MAC as with switch based image. The _eth0_ with _ifconfig_, or _eth0_ and _eth1_ with _ip a_, have different MACs, the origin of which is unclear, and not to be seen on output of switch based image.

adrianschmutzler commented:
While there are a lot of comment, I haven't found a real Tested-by above. I'd happy if some testers could provide those (including the relevant devices).

@dengqf6 I cannot review "
generic: backport mv88e6xxx port mirroring support". To me, this sounds like  an additional feature, so being optional. 
With some Tested-by, I'd consider to merge this without the wifi migration patch and the previously mentioned one if that makes sense?

dengqf6 commented:
> "generic: backport mv88e6xxx port mirroring support". To me, this sounds like an additional feature, so being optional.

I added mirroring support for legacy swconfig driver "mvsw61xx" in commit 18501889f6e353334152b31517bc21094d672da7. After switching to DSA the feature is lost, so I backported support for the DSA counterpart.

anomeome commented:
So appears this PR warrants testing with a 4.19 image, and that _breakage_ is to be anticipated in keeping one's device settings for a couple of config files on a running system. This further complicates the _support_ issues when this gets pushed to _master_. Is there any kind of game plan around that, maybe in conjunction with setting 5.4 as the default kernel version.

borkra commented:
It looks like SQM does not understand this. The interface is wan@eth0 but luci sqm gui displays wan and script tries to interface with 'wan', which does not exist.

lucize commented:
I'll test on wrt3200acm

schm0 commented:
SQM works fine.
The wan interface does exist and sqm scripts can use it just fine.

borkra commented:
If everything works fine, why do I get this every time?

>  24 17:00:52 2020 user.notice SQM: ERROR: cmd_wrapper: tc: FAILURE (2): /usr/sbin/tc qdisc del dev wan root
Sun May 24 17:00:52 2020 user.notice SQM: ERROR: cmd_wrapper: tc: LAST ERROR: RTNETLINK answers: No such file or directory

As far as I can see tc command trying to delete qdisc from interface that does not exist.

anomeome commented:
What exists is apparently dependant on how one looks at what exists (ifconfig, ip a), but SQM (re)starts error free with _wan_ on both a _mamba_ and a _rango_ in my environs.

borkra commented:
> 
> 
> What exists is apparently dependant on how one looks at what exists (ifconfig, ip a), but SQM (re)starts error free with _wan_ on both a _mamba_ and a _rango_ in my environs.

I am looking with ip, from the same package as tc.

borkra commented:
Using cobra the problem is easily reproducible:

In luci uncheck SQM Enable checkbox, Hit Apply and Save button, scroll to the bottom of the System log screen, observe the follwoing:

Mon May 25 13:03:19 2020 user.notice SQM: Stopping SQM on wan
Mon May 25 13:03:19 2020 user.notice SQM: ERROR: cmd_wrapper: tc: FAILURE (2): /usr/sbin/tc qdisc del dev wan root
Mon May 25 13:03:19 2020 user.notice SQM: ERROR: cmd_wrapper: tc: LAST ERROR: RTNETLINK answers: No such file or directory

schm0 commented:
This error is thrown when there was no sqm instance running on that interface. (For what ever reason)

When you enable sqm and run:
tc -s -d qdisc show dev wan
What does the output show?

borkra commented:
> qdisc noqueue 0: root refcnt 2
>  Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0)
>  backlog 0b 0p requeues 0
> qdisc ingress ffff: parent ffff:fff1 ----------------
>  Sent 2363060378 bytes 4134338 pkt (dropped 0, overlimits 0 requeues 0)
>  backlog 0b 0p requeues 0

It should of been cake according to gui, so if it is not enabled, it is the other issue.

anomeome commented:
Not sure if related but I have been seeing this [feature](https://forum.openwrt.org/t/sqm-does-not-always-detect-cake-as-available-qdisc-at-boot/65008)

borkra commented:
Yes, there are a bunch startup ordering issues. I tried to enable/disable way after startup, still the same error. Messages say cake started just fine.

lucize commented:
@adrianschmutzler 

Tested-by Lucian Cristian <lucian.cristian@gmail.com>
tested with 4.14 and 5.4 kernel

SQM works ok on fq_codel and cake 
```
Thu May 28 16:01:55 2020 user.notice SQM: Stopping SQM on br-lan
Thu May 28 16:01:56 2020 user.notice SQM: Starting SQM script: simple.qos on wan, in: 565000 Kbps, out: 551000 Kbps
Thu May 28 16:01:56 2020 user.notice SQM: Using generic sqm_start_default function.
Thu May 28 16:01:56 2020 user.notice SQM: simple.qos was started on wan successfully
Thu May 28 16:03:08 2020 user.notice SQM: Stopping SQM on wan
Thu May 28 16:03:08 2020 user.notice SQM: Starting SQM script: simple.qos on wan, in: 565000 Kbps, out: 551000 Kbps
Thu May 28 16:03:08 2020 user.notice SQM: Using generic sqm_start_default function.
Thu May 28 16:03:08 2020 user.notice SQM: simple.qos was started on wan successfully
```

Speed is better on 5.4 kernel

<details>
<summary>
# DMESG
</summary>


```
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.4.42 (build@debian8) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r13247-631c437)) #0 SMP Thu May 28 13:29:08 2020
[    0.000000] CPU: ARMv7 Processor [414fc091] revision 1 (ARMv7), cr=10c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Linksys WRT3200ACM
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] On node 0 totalpages: 131072
[    0.000000]   Normal zone: 1024 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 131072 pages, LIFO batch:31
[    0.000000] percpu: Embedded 11 pages/cpu s15308 r8192 d21556 u45056
[    0.000000] pcpu-alloc: s15308 r8192 d21556 u45056 alloc=11*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 130048
[    0.000000] Kernel command line: console=ttyS0,115200 root=/dev/mtdblock6 ro rootdelay=1 rootfstype=jffs2 earlyprintk mtdparts=armada-nand:2048K(uboot)ro,128K(u_env),256K(s_env),256K@8064K(devinfo),1920K@8320K(sysdiag),80m@10m(kernel),74m@16m(rootfs),80m@90m(alt_kernel),74m@96m(alt_rootfs),160m@10m(ubifs),-@170m(syscfg)
[    0.000000] Bootloader command line (ignored): console=ttyS0,115200 root=/dev/mtdblock6 ro rootdelay=1 rootfstype=jffs2 earlyprintk mtdparts=armada-nand:2048K(uboot)ro,128K(u_env),256K(s_env),256K@8064K(devinfo),1920K@8320K(sysdiag),80m@10m(kernel),74m@16m(rootfs),80m@90m(alt_kernel),74m@96m(alt_rootfs),160m@10m(ubifs),-@170m(syscfg)
[    0.000000] Dentry cache hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.000000] Inode-cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 509956K/524288K available (6144K kernel code, 195K rwdata, 1608K rodata, 1024K init, 229K bss, 14332K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=4 to nr_cpu_ids=2.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=2
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] L2C-310 enabling early BRESP for Cortex-A9
[    0.000000] L2C-310 full line of zeros enabled for Cortex-A9
[    0.000000] L2C-310 D prefetch enabled, offset 1 lines
[    0.000000] L2C-310 dynamic clock gating enabled, standby mode enabled
[    0.000000] L2C-310 Coherent cache controller enabled, 16 ways, 1024 kB
[    0.000000] L2C-310 Coherent: CACHE_ID 0x410054c9, AUX_CTRL 0x56070001
[    0.000000] random: get_random_bytes called from start_kernel+0x3b0/0x538 with crng_init=0
[    0.000005] sched_clock: 64 bits at 933MHz, resolution 1ns, wraps every 4398046511103ns
[    0.000014] clocksource: arm_global_timer: mask: 0xffffffffffffffff max_cycles: 0x1ae5b571769, max_idle_ns: 881590513431 ns
[    0.000026] Switching to timer-based delay loop, resolution 1ns
[    0.000132] Ignoring duplicate/late registration of read_current_timer delay
[    0.000137] clocksource: armada_370_xp_clocksource: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 76450417870 ns
[    0.000240] Calibrating delay loop (skipped), value calculated using timer frequency.. 1866.00 BogoMIPS (lpj=9330000)
[    0.000245] pid_max: default: 32768 minimum: 301
[    0.000291] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.000296] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.000500] CPU: Testing write buffer coherency: ok
[    0.000510] CPU0: Spectre v2: using BPIALL workaround
[    0.000590] CPU0: thread -1, cpu 0, socket 0, mpidr 80000000
[    0.000725] Setting up static identity map for 0x100000 - 0x100060
[    0.000785] mvebu-soc-id: MVEBU SoC ID=0x6820, Rev=0x4
[    0.000848] mvebu-pmsu: Initializing Power Management Service Unit
[    0.000887] rcu: Hierarchical SRCU implementation.
[    0.000967] smp: Bringing up secondary CPUs ...
[    0.001067] Booting CPU 1
[    0.001204] CPU1: thread -1, cpu 1, socket 0, mpidr 80000001
[    0.001207] CPU1: Spectre v2: using BPIALL workaround
[    0.001248] smp: Brought up 1 node, 2 CPUs
[    0.001251] SMP: Total of 2 processors activated (3732.00 BogoMIPS).
[    0.001254] CPU: All CPU(s) started in SVC mode.
[    0.002553] VFP support v0.3: implementor 41 architecture 3 part 30 variant 9 rev 4
[    0.002603] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.002609] futex hash table entries: 512 (order: 3, 32768 bytes, linear)
[    0.002652] pinctrl core: initialized pinctrl subsystem
[    0.002886] NET: Registered protocol family 16
[    0.003360] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.003949] cpuidle: using governor ladder
[    0.004085] mvebu-pmsu: CPU hotplug support is currently broken on Armada 38x: disabling
[    0.004090] mvebu-pmsu: CPU idle is currently broken on Armada 38x: disabling
[    0.010121] cryptd: max_cpu_qlen set to 1000
[    0.011155] SCSI subsystem initialized
[    0.011218] libata version 3.00 loaded.
[    0.011307] usbcore: registered new interface driver usbfs
[    0.011328] usbcore: registered new interface driver hub
[    0.011345] usbcore: registered new device driver usb
[    0.012198] clocksource: Switched to clocksource arm_global_timer
[    0.012462] thermal_sys: Registered thermal governor 'step_wise'
[    0.012593] NET: Registered protocol family 2
[    0.012832] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.012843] TCP established hash table entries: 4096 (order: 2, 16384 bytes, linear)
[    0.012861] TCP bind hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.012889] TCP: Hash tables configured (established 4096 bind 4096)
[    0.012922] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.012935] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.012998] NET: Registered protocol family 1
[    0.013009] PCI: CLS 0 bytes, default 64
[    0.014215] workingset: timestamp_bits=30 max_order=17 bucket_order=0
[    0.015757] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.015761] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.017791] armada-38x-pinctrl f1018000.pinctrl: registered pinctrl driver
[    0.018578] mvebu-pcie soc:pcie: PCI host bridge to bus 0000:00
[    0.018585] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.018590] pci_bus 0000:00: root bus resource [mem 0xe0000000-0xe7ffffff]
[    0.018593] pci_bus 0000:00: root bus resource [io  0x1000-0xeffff]
[    0.018652] pci 0000:00:01.0: [11ab:6820] type 01 class 0x060400
[    0.018666] pci 0000:00:01.0: reg 0x38: [mem 0x00000000-0x000007ff pref]
[    0.018771] pci 0000:00:02.0: [11ab:6820] type 01 class 0x060400
[    0.018781] pci 0000:00:02.0: reg 0x38: [mem 0x00000000-0x000007ff pref]
[    0.019424] PCI: bus0: Fast back to back transfers disabled
[    0.019429] pci 0000:00:01.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    0.019435] pci 0000:00:02.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    0.019480] pci 0000:01:00.0: [11ab:2b40] type 00 class 0x020000
[    0.019503] pci 0000:01:00.0: reg 0x10: [mem 0x40000000-0x401fffff 64bit pref]
[    0.019513] pci 0000:01:00.0: reg 0x18: [mem 0x40200000-0x402fffff 64bit pref]
[    0.019577] pci 0000:01:00.0: supports D1 D2
[    0.019581] pci 0000:01:00.0: PME# supported from D0 D1 D3hot D3cold
[    0.020288] PCI: bus1: Fast back to back transfers disabled
[    0.020294] pci_bus 0000:01: busn_res: [bus 01-ff] end is updated to 01
[    0.020344] pci 0000:02:00.0: [11ab:2b40] type 00 class 0x020000
[    0.020372] pci 0000:02:00.0: reg 0x10: [mem 0x42000000-0x421fffff 64bit pref]
[    0.020385] pci 0000:02:00.0: reg 0x18: [mem 0x42200000-0x422fffff 64bit pref]
[    0.020473] pci 0000:02:00.0: supports D1 D2
[    0.020476] pci 0000:02:00.0: PME# supported from D0 D1 D3hot D3cold
[    0.020497] pci 0000:02:00.0: 2.000 Gb/s available PCIe bandwidth, limited by 2.5 GT/s x1 link at 0000:00:02.0 (capable of 4.000 Gb/s with 5 GT/s x1 link)
[    0.021166] PCI: bus2: Fast back to back transfers disabled
[    0.021171] pci_bus 0000:02: busn_res: [bus 02-ff] end is updated to 02
[    0.021194] pci 0000:00:01.0: BAR 8: assigned [mem 0xe0000000-0xe02fffff]
[    0.021199] pci 0000:00:02.0: BAR 8: assigned [mem 0xe0400000-0xe06fffff]
[    0.021204] pci 0000:00:01.0: BAR 6: assigned [mem 0xe0300000-0xe03007ff pref]
[    0.021209] pci 0000:00:02.0: BAR 6: assigned [mem 0xe0700000-0xe07007ff pref]
[    0.021215] pci 0000:01:00.0: BAR 0: assigned [mem 0xe0000000-0xe01fffff 64bit pref]
[    0.021225] pci 0000:01:00.0: BAR 2: assigned [mem 0xe0200000-0xe02fffff 64bit pref]
[    0.021238] pci 0000:00:01.0: PCI bridge to [bus 01]
[    0.021246] pci 0000:00:01.0:   bridge window [mem 0xe0000000-0xe02fffff]
[    0.021253] pci 0000:02:00.0: BAR 0: assigned [mem 0xe0400000-0xe05fffff 64bit pref]
[    0.021263] pci 0000:02:00.0: BAR 2: assigned [mem 0xe0600000-0xe06fffff 64bit pref]
[    0.021272] pci 0000:00:02.0: PCI bridge to [bus 02]
[    0.021276] pci 0000:00:02.0:   bridge window [mem 0xe0400000-0xe06fffff]
[    0.021492] mv_xor f1060800.xor: Marvell shared XOR driver
[    0.082160] mv_xor f1060800.xor: Marvell XOR (Descriptor Mode): ( xor cpy intr )
[    0.082277] mv_xor f1060900.xor: Marvell shared XOR driver
[    0.142115] mv_xor f1060900.xor: Marvell XOR (Descriptor Mode): ( xor cpy intr )
[    0.142270] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.142520] printk: console [ttyS0] disabled
[    0.162592] f1012000.serial: ttyS0 at MMIO 0xf1012000 (irq = 22, base_baud = 12500000) is a 16550A
[    1.019937] printk: console [ttyS0] enabled
[    1.025495] loop: module loaded
[    1.028862] ahci-mvebu f10a8000.sata: f10a8000.sata supply ahci not found, using dummy regulator
[    1.037723] ahci-mvebu f10a8000.sata: f10a8000.sata supply phy not found, using dummy regulator
[    1.046486] ahci-mvebu f10a8000.sata: f10a8000.sata supply target not found, using dummy regulator
[    1.055553] ahci-mvebu f10a8000.sata: AHCI 0001.0000 32 slots 2 ports 6 Gbps 0x3 impl platform mode
[    1.064648] ahci-mvebu f10a8000.sata: flags: 64bit ncq sntf led only pmp fbs pio slum part sxs 
[    1.073741] scsi host0: ahci-mvebu
[    1.077344] scsi host1: ahci-mvebu
[    1.080835] ata1: SATA max UDMA/133 mmio [mem 0xf10a8000-0xf10a9fff] port 0x100 irq 41
[    1.088800] ata2: SATA max UDMA/133 mmio [mem 0xf10a8000-0xf10a9fff] port 0x180 irq 41
[    1.097222] nand: device found, Manufacturer ID: 0x01, Chip ID: 0xda
[    1.103614] nand: AMD/Spansion S34ML02G2
[    1.107553] nand: 256 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 128
[    1.115589] Bad block table found at page 131008, version 0x01
[    1.122084] Bad block table found at page 130944, version 0x01
[    1.128355] 11 fixed-partitions partitions found on MTD device pxa3xx_nand-0
[    1.135443] Creating 11 MTD partitions on "pxa3xx_nand-0":
[    1.140952] 0x000000000000-0x000000200000 : "u-boot"
[    1.146189] 0x000000200000-0x000000220000 : "u_env"
[    1.151278] 0x000000220000-0x000000260000 : "s_env"
[    1.156371] 0x0000007e0000-0x000000820000 : "devinfo"
[    1.161637] 0x000000820000-0x000000a00000 : "sysdiag"
[    1.166928] 0x000000a00000-0x000005a00000 : "kernel1"
[    1.172302] 0x000001000000-0x000005a00000 : "ubi"
[    1.177304] 0x000005a00000-0x00000aa00000 : "kernel2"
[    1.182668] 0x000006000000-0x00000aa00000 : "rootfs2"
[    1.188022] 0x00000aa00000-0x000010000000 : "syscfg"
[    1.193309] 0x000000260000-0x000000820000 : "unused_area"
[    1.199217] libphy: Fixed MDIO Bus: probed
[    1.203583] orion-mdio f1072004.mdio: IRQ index 0 not found
[    1.209217] libphy: orion_mdio_bus: probed
[    1.213456] mv88e6085 f1072004.mdio-mii:00: switch 0x3520 detected: Marvell 88E6352, revision 1
[    1.384856] libphy: mv88e6xxx SMI: probed
[    1.390636] mvneta_bm f10c8000.bm: Buffer Manager for network controller enabled
[    1.399329] mvneta f1070000.ethernet eth0: Using hardware mac address xx:xx:xx:xx:xx:xx
[    1.407440] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    1.413999] ehci-platform: EHCI generic platform driver
[    1.419306] ehci-orion: EHCI orion driver
[    1.423404] orion-ehci f1058000.usb: EHCI Host Controller
[    1.428834] orion-ehci f1058000.usb: new USB bus registered, assigned bus number 1
[    1.433469] ata1: SATA link down (SStatus 0 SControl 300)
[    1.436478] orion-ehci f1058000.usb: irq 38, io mem 0xf1058000
[    1.441884] ata2: SATA link down (SStatus 0 SControl 300)
[    1.471795] orion-ehci f1058000.usb: USB 2.0 started, EHCI 1.00
[    1.477946] hub 1-0:1.0: USB hub found
[    1.481737] hub 1-0:1.0: 1 port detected
[    1.486029] usbcore: registered new interface driver usb-storage
[    1.492206] i2c /dev entries driver
[    1.498098] watchdog: f1020300.watchdog: driver supplied timeout (4294967295) out of range
[    1.506407] watchdog: f1020300.watchdog: falling back to default timeout (171)
[    1.513737] orion_wdt: Initial timeout 171 sec
[    1.518405] sdhci: Secure Digital Host Controller Interface driver
[    1.524615] sdhci: Copyright(c) Pierre Ossman
[    1.560657] mmc0: SDHCI controller on f10d8000.sdhci [f10d8000.sdhci] using ADMA
[    1.568175] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.577701] marvell-cesa f1090000.crypto: CESA device successfully registered
[    1.585285] NET: Registered protocol family 10
[    1.590091] Segment Routing with IPv6
[    1.593805] NET: Registered protocol family 17
[    1.598280] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    1.611330] 8021q: 802.1Q VLAN Support v1.8
[    1.615589] Registering SWP/SWPB emulation handler
[    1.622633] mv88e6085 f1072004.mdio-mii:00: switch 0x3520 detected: Marvell 88E6352, revision 1
[    1.649253] random: fast init done
[    1.653710] mmc0: new high speed SDIO card at address 0001
[    1.801069] libphy: mv88e6xxx SMI: probed
[    1.973682] mv88e6085 f1072004.mdio-mii:00 lan4 (uninitialized): PHY [mv88e6xxx-1:00] driver [Marvell 88E1540]
[    1.990983] mv88e6085 f1072004.mdio-mii:00 lan3 (uninitialized): PHY [mv88e6xxx-1:01] driver [Marvell 88E1540]
[    2.008229] mv88e6085 f1072004.mdio-mii:00 lan2 (uninitialized): PHY [mv88e6xxx-1:02] driver [Marvell 88E1540]
[    2.025476] mv88e6085 f1072004.mdio-mii:00 lan1 (uninitialized): PHY [mv88e6xxx-1:03] driver [Marvell 88E1540]
[    2.040660] mv88e6085 f1072004.mdio-mii:00 wan (uninitialized): PHY [mv88e6xxx-1:04] driver [Marvell 88E1540]
[    2.071090] mv88e6085 f1072004.mdio-mii:00: configuring for fixed/ link mode
[    2.079453] DSA: tree 0 setup
[    2.082592] xhci-hcd f10f8000.usb3: xHCI Host Controller
[    2.087935] xhci-hcd f10f8000.usb3: new USB bus registered, assigned bus number 2
[    2.095514] xhci-hcd f10f8000.usb3: hcc params 0x0a000990 hci version 0x100 quirks 0x0000000000010010
[    2.104795] xhci-hcd f10f8000.usb3: irq 44, io mem 0xf10f8000
[    2.110879] hub 2-0:1.0: USB hub found
[    2.114669] hub 2-0:1.0: 1 port detected
[    2.118699] xhci-hcd f10f8000.usb3: xHCI Host Controller
[    2.124044] xhci-hcd f10f8000.usb3: new USB bus registered, assigned bus number 3
[    2.131567] xhci-hcd f10f8000.usb3: Host supports USB 3.0 SuperSpeed
[    2.137978] usb usb3: We don't know the algorithms for LPM for this host, disabling LPM.
[    2.146255] hub 3-0:1.0: USB hub found
[    2.150033] hub 3-0:1.0: 1 port detected
[    2.154473] UBI: auto-attach mtd6
[    2.157802] ubi0: attaching mtd6
[    2.162718] mv88e6085 f1072004.mdio-mii:00: Link is Up - 1Gbps/Full - flow control off
[    2.551397] ubi0: scanning is finished
[    2.562860] ubi0: attached mtd6 (name "ubi", size 74 MiB)
[    2.568282] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    2.575192] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    2.582012] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    2.589002] ubi0: good PEBs: 592, bad PEBs: 0, corrupted PEBs: 0
[    2.595037] ubi0: user volume: 2, internal volumes: 1, max. volumes count: 128
[    2.602292] ubi0: max/mean erase counter: 11/6, WL threshold: 4096, image sequence number: 552670511
[    2.611464] ubi0: available PEBs: 0, total reserved PEBs: 592, PEBs reserved for bad PEB handling: 40
[    2.621145] ubi0: background thread "ubi_bgt0d" started, PID 879
[    2.621249] block ubiblock0_0: created from ubi0:0(rootfs)
[    2.632695] ubiblock: device ubiblock0_0 (rootfs) set to be root filesystem
[    2.639687] hctosys: unable to open rtc device (rtc0)
[    2.644962] Waiting 1 sec before mounting root device...
[    3.683789] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    3.691318] Freeing unused kernel memory: 1024K
[    3.731887] Run /sbin/init as init process
[    3.901848] init: Console is alive
[    3.905335] init: - watchdog -
[    4.368137] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.391204] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.402914] init: - preinit -
[    4.733841] random: jshn: uninitialized urandom read (4 bytes read)
[    4.768382] random: jshn: uninitialized urandom read (4 bytes read)
[    4.844770] random: jshn: uninitialized urandom read (4 bytes read)
[    9.040957] mount_root: loading kmods from internal overlay
[    9.053847] kmodloader: loading kernel modules from //etc/modules-boot.d/*
[    9.061856] kmodloader: done loading kernel modules from //etc/modules-boot.d/*
[    9.649927] UBIFS (ubi0:1): Mounting in unauthenticated mode
[    9.655673] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 982
[    9.757088] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[    9.764958] UBIFS (ubi0:1): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[    9.774919] UBIFS (ubi0:1): FS size: 53329920 bytes (50 MiB, 420 LEBs), journal size 2666496 bytes (2 MiB, 21 LEBs)
[    9.785404] UBIFS (ubi0:1): reserved for root: 2518903 bytes (2459 KiB)
[    9.792049] UBIFS (ubi0:1): media format: w4/r0 (latest is w5/r0), UUID 0FA02B95-FDF3-467C-B855-728F45ED4262, small LPT model
[    9.804124] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[    9.811151] block: unable to load configuration (fstab: Entry not found)
[    9.817901] block: attempting to load /tmp/ubifs_cfg/etc/config/fstab
[    9.824395] block: unable to load configuration (fstab: Entry not found)
[    9.831137] block: attempting to load /etc/config/fstab
[    9.836792] block: unable to load configuration (fstab: Entry not found)
[    9.843541] block: no usable configuration
[    9.850331] UBIFS (ubi0:1): un-mount UBI device 0
[    9.855071] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" stops
[    9.863625] UBIFS (ubi0:1): Mounting in unauthenticated mode
[    9.871806] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 984
[    9.970777] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[    9.978653] UBIFS (ubi0:1): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[    9.988616] UBIFS (ubi0:1): FS size: 53329920 bytes (50 MiB, 420 LEBs), journal size 2666496 bytes (2 MiB, 21 LEBs)
[    9.999100] UBIFS (ubi0:1): reserved for root: 2518903 bytes (2459 KiB)
[   10.005749] UBIFS (ubi0:1): media format: w4/r0 (latest is w5/r0), UUID 0FA02B95-FDF3-467C-B855-728F45ED4262, small LPT model
[   10.547159] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   10.554192] block: unable to load configuration (fstab: Entry not found)
[   10.560937] block: attempting to load /tmp/ubifs_cfg/etc/config/fstab
[   10.567437] block: unable to load configuration (fstab: Entry not found)
[   10.574183] block: attempting to load /etc/config/fstab
[   10.579449] block: unable to load configuration (fstab: Entry not found)
[   10.586191] block: no usable configuration
[   10.590630] mount_root: overlay filesystem has not been fully initialized yet
[   10.598254] mount_root: switching to ubifs overlay
[   10.611879] overlayfs: upper fs does not support xattr, falling back to index=off and metacopy=off.
[   10.718933] ubi1: attaching mtd9
[   11.162579] ubi1: scanning is finished
[   11.174175] ubi1: attached mtd9 (name "syscfg", size 86 MiB)
[   11.179860] ubi1: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[   11.186776] ubi1: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[   11.193598] ubi1: VID header offset: 2048 (aligned 2048), data offset: 4096
[   11.200589] ubi1: good PEBs: 680, bad PEBs: 8, corrupted PEBs: 0
[   11.206623] ubi1: user volume: 1, internal volumes: 1, max. volumes count: 128
[   11.213879] ubi1: max/mean erase counter: 9/5, WL threshold: 4096, image sequence number: 2139766255
[   11.223055] ubi1: available PEBs: 0, total reserved PEBs: 680, PEBs reserved for bad PEB handling: 32
[   11.232323] ubi1: background thread "ubi_bgt1d" started, PID 998
[   11.243083] UBIFS (ubi1:0): Mounting in unauthenticated mode
[   11.249667] UBIFS (ubi1:0): background thread "ubifs_bgt1_0" started, PID 1002
[   11.335640] UBIFS (ubi1:0): UBIFS: mounted UBI device 1, volume 0, name "syscfg"
[   11.343078] UBIFS (ubi1:0): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[   11.353040] UBIFS (ubi1:0): FS size: 80375808 bytes (76 MiB, 633 LEBs), journal size 4063232 bytes (3 MiB, 32 LEBs)
[   11.363524] UBIFS (ubi1:0): reserved for root: 3796347 bytes (3707 KiB)
[   11.370167] UBIFS (ubi1:0): media format: w4/r0 (latest is w5/r0), UUID E8CEBF95-5EC1-44DE-97F8-FD66A023B0AD, small LPT model
[   11.382929] urandom-seed: Seed file not found (/etc/urandom.seed)
[   11.414594] procd: - early -
[   11.417507] procd: - watchdog -
[   11.945626] procd: - watchdog -
[   11.948852] procd: - ubus -
[   11.975952] urandom_read: 3 callbacks suppressed
[   11.975955] random: ubusd: uninitialized urandom read (4 bytes read)
[   12.001129] random: ubusd: uninitialized urandom read (4 bytes read)
[   12.007589] random: ubusd: uninitialized urandom read (4 bytes read)
[   12.014187] procd: - init -
[   12.176212] kmodloader: loading kernel modules from /etc/modules.d/*
[   12.220274] urngd: v1.0.2 started.
[   12.260333] random: crng init done
[   12.263785] random: 1 urandom warning(s) missed due to ratelimiting
[   12.327583] NET: Registered protocol family 38
[   12.349402] ntfs: driver 2.1.32 [Flags: R/O MODULE].
[   12.357377] NET: Registered protocol family 15
[   12.366757] Initializing XFRM netlink socket
[   12.377805] l2tp_core: L2TP core driver, V2.0
[   12.382828] l2tp_netlink: L2TP netlink interface
[   12.387999] IPv4 over IPsec tunneling driver
[   12.395278] gre: GRE over IPv4 demultiplexor driver
[   12.400746] ip_gre: GRE over IPv4 tunneling driver
[   12.440414] ip6_gre: GRE over IPv6 tunneling driver
[   12.463404] cryptodev: driver 1.10 loaded.
[   12.469167] Loading modules backported from Linux version v5.7-rc3-0-g6a8b55ed4056
[   12.476789] Backport generated by backports.git v5.7-rc3-1-0-gc0c7d2bb
[   12.522141] <<Marvell Mac80211 Wireless PCIE Network Driver version 10.3.8.0-20181210>>
[   12.530195] pci 0000:00:01.0: enabling device (0140 -> 0142)
[   12.535994] ieee80211 phy0: priv->pcmd_buf = d4f258b5  priv->pphys_cmd_buf = 4aaca072
[   12.660550] ieee80211 phy0: fw download start
[   12.660555] ieee80211 phy0: ddr init: download start
[   12.660935] ieee80211 phy0: ddr init: download complete
[   12.804731] ieee80211 phy0: FwSize = 451040 downloaded Size = 451040 curr_iteration 65524
[   15.671793] ieee80211 phy0: fw download complete
[   15.671846] ieee80211 phy0: device node: pcie@1,0
[   15.687196] ieee80211 phy0: pcmd = d4f258b5
[   15.706498] ieee80211 phy0: firmware version: 0x903020c
[   16.011790] ieee80211 phy0: firmware region code: 30
[   16.049456] ieee80211 phy0: 2G disabled, 5G enabled
[   16.054370] ieee80211 phy0: 4 TX antennas, 4 RX antennas
[   16.059852] pci 0000:00:02.0: enabling device (0140 -> 0142)
[   16.065653] ieee80211 phy1: priv->pcmd_buf = e68f2df9  priv->pphys_cmd_buf = fc86264d
[   16.071821] ieee80211 phy1: fw download start
[   16.071827] ieee80211 phy1: ddr init: download start
[   16.072209] ieee80211 phy1: ddr init: download complete
[   16.213845] ieee80211 phy1: FwSize = 451040 downloaded Size = 451040 curr_iteration 65527
[   17.407797] ieee80211 phy1: fw download complete
[   17.407841] ieee80211 phy1: device node: pcie@2,0
[   17.423153] ieee80211 phy1: pcmd = e68f2df9
[   17.440813] ieee80211 phy1: firmware version: 0x903020c
[   17.745800] ieee80211 phy1: firmware region code: 30
[   17.778607] ieee80211 phy1: 2G enabled, 5G disabled
[   17.783546] ieee80211 phy1: 4 TX antennas, 4 RX antennas
[   17.800857] hidraw: raw HID events driver (C) Jiri Kosina
[   17.806907] Mirror/redirect action on
[   17.813471] u32 classifier
[   17.816194]     input device check on
[   17.819873]     Actions configured
[   17.830788] Bluetooth: Core ver 2.22
[   17.834449] NET: Registered protocol family 31
[   17.838916] Bluetooth: HCI device and connection manager initialized
[   17.845355] Bluetooth: HCI socket layer initialized
[   17.850271] Bluetooth: L2CAP socket layer initialized
[   17.855370] Bluetooth: SCO socket layer initialized
[   17.863125] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   17.868465] Bluetooth: BNEP filters: protocol multicast
[   17.873761] Bluetooth: BNEP socket layer initialized
[   17.879323] Bridge firewalling registered
[   17.885344] Bluetooth: vendor=0x2df, device=0x9136, class=255, fn=2
[   18.621245] Bluetooth: FW download over, size 616840 bytes
[   19.231822] btmrvl_sdio mmc0:0001:2: sdio device tree data not available
[   19.239433] usbcore: registered new interface driver btusb
[   19.245370] usbcore: registered new interface driver cdc_wdm
[   19.251771] Bluetooth: HCI UART driver ver 2.3
[   19.256244] Bluetooth: HCI UART protocol H4 registered
[   19.261403] Bluetooth: HCI UART protocol BCSP registered
[   19.267219] Bluetooth: HIDP (Human Interface Emulation) ver 1.2
[   19.273183] Bluetooth: HIDP socket layer initialized
[   19.284336] usbcore: registered new interface driver r8152
[   19.290590] Bluetooth: RFCOMM TTY layer initialized
[   19.295508] Bluetooth: RFCOMM socket layer initialized
[   19.300675] Bluetooth: RFCOMM ver 1.11
[   19.307618] wireguard: WireGuard 1.0.20200520 loaded. See www.wireguard.com for information.
[   19.316112] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[   19.332579] IPsec XFRM device driver
[   19.342249] xt_time: kernel timezone is -0000
[   19.347351] usbcore: registered new interface driver cdc_ncm
[   19.353318] usbcore: registered new interface driver cdc_subset
[   19.359785] usbcore: registered new interface driver dm9601
[   19.365696] usbcore: registered new interface driver huawei_cdc_ncm
[   19.379446] PPP generic driver version 2.4.2
[   19.392438] PPP MPPE Compression module registered
[   19.398248] NET: Registered protocol family 24
[   19.404088] usbcore: registered new interface driver qmi_wwan
[   19.415694] usbcore: registered new interface driver cdc_mbim
[   19.422683] l2tp_ppp: PPPoL2TP kernel driver, V2.0
[   19.429120] kmodloader: done loading kernel modules from /etc/modules.d/*
[   21.525155] mvneta f1070000.ethernet eth0: configuring for fixed/rgmii-id link mode
[   21.533291] mvneta f1070000.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[   21.541166] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   21.551464] mv88e6085 f1072004.mdio-mii:00 lan1: configuring for phy/gmii link mode
[   21.562311] 8021q: adding VLAN 0 to HW filter on device lan1
[   21.569151] br-lan: port 1(lan1) entered blocking state
[   21.574442] br-lan: port 1(lan1) entered disabled state
[   21.584839] device lan1 entered promiscuous mode
[   21.589479] device eth0 entered promiscuous mode
[   21.611749] mv88e6085 f1072004.mdio-mii:00 lan2: configuring for phy/gmii link mode
[   21.633748] 8021q: adding VLAN 0 to HW filter on device lan2
[   21.639589] br-lan: port 2(lan2) entered blocking state
[   21.644886] br-lan: port 2(lan2) entered disabled state
[   21.661518] device lan2 entered promiscuous mode
[   21.700967] mv88e6085 f1072004.mdio-mii:00 lan3: configuring for phy/gmii link mode
[   21.768594] 8021q: adding VLAN 0 to HW filter on device lan3
[   21.778661] br-lan: port 3(lan3) entered blocking state
[   21.783957] br-lan: port 3(lan3) entered disabled state
[   21.796237] device lan3 entered promiscuous mode
[   21.819303] mv88e6085 f1072004.mdio-mii:00 lan4: configuring for phy/gmii link mode
[   21.879826] 8021q: adding VLAN 0 to HW filter on device lan4
[   21.889459] br-lan: port 4(lan4) entered blocking state
[   21.894728] br-lan: port 4(lan4) entered disabled state
[   21.902557] device lan4 entered promiscuous mode
[   21.924343] mv88e6085 f1072004.mdio-mii:00 wan: configuring for phy/gmii link mode
[   21.952071] 8021q: adding VLAN 0 to HW filter on device wan
[   22.564949] ieee80211 phy0: change: 0xffffffff
[   22.670412] br-lan: port 5(wlan0) entered blocking state
[   22.675763] br-lan: port 5(wlan0) entered disabled state
[   22.681178] device wlan0 entered promiscuous mode
[   22.685992] br-lan: port 5(wlan0) entered blocking state
[   22.691330] br-lan: port 5(wlan0) entered forwarding state
[   22.697543] br-lan: port 5(wlan0) entered disabled state
[   22.751889] device wlan0 left promiscuous mode
[   22.756412] br-lan: port 5(wlan0) entered disabled state
[   24.010849] ieee80211 phy0: change: 0xffffffff
[   24.148543] br-lan: port 5(wlan0) entered blocking state
[   24.153903] br-lan: port 5(wlan0) entered disabled state
[   24.159422] device wlan0 entered promiscuous mode
[   25.321112] mv88e6085 f1072004.mdio-mii:00 lan3: Link is Up - 1Gbps/Full - flow control rx/tx
[   25.331109] br-lan: port 3(lan3) entered blocking state
[   25.336374] br-lan: port 3(lan3) entered forwarding state
[   25.346332] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   25.696624] mv88e6085 f1072004.mdio-mii:00 wan: Link is Up - 1Gbps/Full - flow control rx/tx
[   25.705968] IPv6: ADDRCONF(NETDEV_CHANGE): wan: link becomes ready

```

</details>

adrianschmutzler commented:
Thanks, so the question is whether mvebu can/should be switched to 5.4 before adding this?

lucize commented:
from my quick test on iperf3 it seems that 5.4 is a bit speedier, I can use it like this some days and see the stability, but don't know what can go wrong

adrianschmutzler commented:
Just send a patch for switching to 5.4 to the mailing list, as the recent reports in the 5.4 PR have all been positive. Depending on feedback I'd then merge the kernel bump and this afterwards if no new issues show up.

borkra commented:
> Thu May 28 16:01:55 2020 user.notice SQM: Stopping SQM on br-lan
> Thu May 28 16:01:56 2020 user.notice SQM: Starting SQM script: simple.qos on wan, in: 565000 Kbps, out: 551000 Kbps
> Thu May 28 16:01:56 2020 user.notice SQM: Using generic sqm_start_default function.
> Thu May 28 16:01:56 2020 user.notice SQM: simple.qos was started on wan successfully
> Thu May 28 16:03:08 2020 user.notice SQM: Stopping SQM on wan
> Thu May 28 16:03:08 2020 user.notice SQM: Starting SQM script: simple.qos on wan, in: 565000 Kbps, out: 551000 Kbps
> Thu May 28 16:03:08 2020 user.notice SQM: Using generic sqm_start_default function.
> Thu May 28 16:03:08 2020 user.notice SQM: simple.qos was started on wan successfully

This test was performed on 'simple.qos', not on 'layered_cake.qos' or 'piece_of_cake.qos', which are required for cake. And why exactly the conclusion that everything works correctly with cake when the cake test case was not even tested?

lucize commented:
don't get offended :)

please
```

Thu May 28 19:45:39 2020 user.notice SQM: Starting SQM script: piece_of_cake.qos on wan, in: 585000 Kbps, out: 510000 Kbps
Thu May 28 19:45:39 2020 user.notice SQM: Using generic sqm_start_default function.
Thu May 28 19:45:39 2020 user.notice SQM: piece_of_cake.qos was started on wan successfully
```
![image](https://user-images.githubusercontent.com/6313831/83186492-43241c00-a135-11ea-8366-eeff3b2cc1d7.png)

![image](https://user-images.githubusercontent.com/6313831/83186518-4ddeb100-a135-11ea-8c0e-0f0d7846c4e6.png)

![image](https://user-images.githubusercontent.com/6313831/83186545-5931dc80-a135-11ea-8fc1-180e85e8335b.png)

lucize commented:
```
Thu May 28 19:49:19 2020 user.notice SQM: Stopping SQM on wan
Thu May 28 19:49:19 2020 user.notice SQM: Starting SQM script: layer_cake.qos on wan, in: 585000 Kbps, out: 510000 Kbps
Thu May 28 19:49:19 2020 user.notice SQM: Using generic sqm_start_default function.
Thu May 28 19:49:19 2020 user.notice SQM: layer_cake.qos was started on wan successfully

```
did you tried with 5.4 kernel ?

borkra commented:
Yes, I am only using 5.4 kernel, ok, so it starts correctly why does it stop with error then?

Is it just a script error?

adrianschmutzler commented:
Thanks, I've merged this after the 5.4 bump.
(I only removed the Wifi migration patch, will send that to the list and merge it later if there is no objection.)

neheb commented:
@dengqf6 patch 310 looks like it should be upstreamed.
